internal/dag/policy.go

```diff
@@ -203,12 +214,26 @@ func headersPolicyRoute(policy *contour_api_v1.HeadersPolicy, allowHostRewrite b
 	}
 
 	return &HeadersPolicy{
-		Set:         set,
-		HostRewrite: hostRewrite,
-		Remove:      rl,
+		Set:               set,
+		HostRewrite:       hostRewrite,
+		HostRewriteHeader: hostRewriteHeader,
+		Remove:            rl,
 	}, nil
 }
 
+// extractHostRewriteHeaderValue returns the value of the header
+func extractHostRewriteHeaderValue(s string) string {
+	// match "%REQ(<X-Foo-Bar>)%"
+	re := regexp.MustCompile(`%REQ\((\S+)\)%`)
```

Comment:
this can panic and can be expensive so I think we want it outside of the datapath